ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh

```diff
@@ -445,31 +449,74 @@ EOF
   done
 }
 
-# inject_spot_instance_config is an AWS specific option that enables the use of AWS spot instances for worker nodes
+# inject_spot_instance_config is an AWS specific option that enables the
+# use of AWS spot instances.
+# PARAMS:
+# $1: Path to base output directory of `openshift-install create manifests`
+# $2: Either "workers" or "masters" to enable spot instances on the
+#     compute or control machines, respectively.
 function inject_spot_instance_config() {
   local dir=${1}
+  local mtype=${2}
 
-  if [ ! -f /tmp/yq ]; then
-    curl -L "https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_$( get_arch )" \
-    -o /tmp/yq && chmod +x /tmp/yq
-  fi
 
-  PATCH="${SHARED_DIR}/machinesets-spot-instances.yaml.patch"
-  cat > "${PATCH}" << EOF
-spec:
-  template:
-    spec:
-      providerSpec:
-        value:
-          spotMarketOptions: {}
-EOF
+  get_yq
 
-  for MACHINESET in $dir/openshift/99_openshift-cluster-api_worker-machineset-*.yaml; do
-    /tmp/yq m -x -i "${MACHINESET}" "${PATCH}"
-    echo "Patched spotMarketOptions into ${MACHINESET}"
+  # Find manifest files
+  local manifests=
+  case "${mtype}" in
+    masters)
+      # Spot masters only works for CAPI today. Validate.
+      if ! [[ "$(/tmp/yq r "${SHARED_DIR}/install-config.yaml" 'featureGates[*]')" == *'ClusterAPIInstall=true'* ]]; then
```

Comment:
We recently introduced platform-specific feature gates [1], so we'll have to check for that one too: `ClusterAPIInstallAWS`.

[1] https://github.com/openshift/api/pull/1868